client/src/document/index.tsx

```diff
@@ -366,3 +382,83 @@ function LoadingError({ error }) {
     </div>
   );
 }
+
+interface FlatFlaw {
+  name: string;
+  flaws: string[];
+  count: number;
+}
+
+function ToggleDocmentFlaws({ doc }: { doc: Doc }) {
+  const { flaws }: { flaws: object } = doc;
+  const [show, toggle] = useReducer((v) => !v, false);
+
+  useEffect(() => {
+    if (window.location.hash && window.location.hash === "#show-flaws") {
+      toggle();
+    }
+  }, []);
+
+  useEffect(() => {
+    if (show) {
+      if (!(window.location.hash && window.location.hash === "#show-flaws")) {
+        window.location.hash = "#show-flaws";
+      }
+    } else {
+      if (window.location.hash && window.location.hash === "#show-flaws") {
+        window.location.hash = "";
+      }
+    }
+  }, [show]);
+
+  let flatFlaws: FlatFlaw[] = [];
+  for (const [name, actualFlaws] of Object.entries(flaws)) {
+    flatFlaws.push({
+      name,
+      flaws: actualFlaws,
+      count: actualFlaws.length,
+    });
+  }
+  flatFlaws.sort((a, b) => b.count - a.count);
```

Comment:
```suggestion
  const flatFlaws: FlatFlaw[] = Object.entries(flaws)
    .map(([name, actualFlaws]) => ({
      name,
      flaws: actualFlaws,
      count: actualFlaws.length,
    }))
    .sort((a, b) => b.count - a.count);
```
Call me a LISPer but I find this way to be more expressive